subprojects/core/src/main/java/org/gradle/internal/service/scopes/ProjectScopeServices.java

```diff
@@ -115,23 +119,27 @@
 public class ProjectScopeServices implements ServiceRegistrationProvider {
 
     public static CloseableServiceRegistry create(
-        ServiceRegistry parent,
+        ServiceRegistry buildServices,
         ProjectInternal project,
         Factory<LoggingManagerInternal> loggingManagerInternalFactory
     ) {
+        BuildLayout buildLayout = buildServices.get(BuildLayout.class);
```

Comment:
🤔  We should avoid direct lookup of services, and instead wire everything via injection. It might not always be possible due to the shortcomings of our framework or the current Gradle architecture, but we should double check that before doing a direct lookup.

Direct lookups hint at the leaking abstraction of service scopes (if they can't be avoided). Then, instead of going around the leakage, we should fix it.

Though, I am pretty certain in this case we should be able to do away with a regular injection.